File: lec06/peptest.java
import java.util.*;
class linearSearch{
    static Scanner scn= new Scanner(System.in);
    public static void input(int[] arr){
        for(int i=0; i<arr.length; i++){
            arr[i]= scn.nextInt();
        }
    }

    


    public static void main(String[] args){
        int size= scn.nextInt();
        int[] arr= new int[size];
        input(arr);

        int max= INT_MIN;

        for(int i=0; i<arr.length; i++){
        
            for(int j=i; j<arr.length; j++){
                int sum=0;
                for(int k=i; k<=j; k++){
                    sum+=arr[k];
                    if(sum>max){
                        max=sum;
                    }
                }
            
            }
        }

    }
}

File: lec06/peptest.cpp
#include <iostream>
using namespace std;
int main(){
    int n;
    cin>> n;
    
    int arr[n];
    for(int i=0; i<n; i++){
        cin>> arr[i];
    }
     int max= INT_MIN;

        for(int i=0; i<sizeof(arr); i++){
        
            for(int j=i; j<sizeof(arr); j++){
                int sum=0;
                for(int k=i; k<=j; k++){
                    sum+=arr[k];
                }
                if(sum>max){
                        max=sum;
                    }
            
            }
        }
        cout << max;
    return 0;
}

